tsdb/tsi1/file_set.go

```diff
@@ -403,6 +403,39 @@ func (fs *FileSet) TagValueSeriesIDIterator(name, key, value []byte) (tsdb.Serie
 	return tsdb.NewSeriesIDSetIterator(ss), nil
 }
 
+// Stats computes aggregate measurement cardinality stats from the raw index data.
+func (fs *FileSet) Stats() MeasurementCardinalityStats {
+	stats := make(MeasurementCardinalityStats)
+	mitr := fs.MeasurementIterator()
+	if mitr == nil {
+		return stats
+	}
+
+	for {
+		// Iterate over each measurement and set cardinality.
+		mm := mitr.Next()
+		if mm == nil {
+			return stats
+		}
+
+		// Obtain all series for measurement.
+		sitr := fs.MeasurementSeriesIDIterator(mm.Name())
+		if sitr == nil {
+			continue
+		}
+
+		// All iterators should be series id set iterators except legacy 1.x data.
```

Comment:
Should it be possible to have legacy 1.x data?

Reply:
We could but it's going to take a long time to compute. This is specifically pre-roaring 1.x. Later 1.x data works ok.

Reply:
No, I mean, there shouldn't be any legacy TSI data in a running 2.x instance.

Reply:
Ah, yes, that's correct. We could panic here instead but I figured just skipping it would be better.